elytron/src/main/java/org/wildfly/extension/elytron/ProviderDefinitions.java

```diff
@@ -199,12 +202,20 @@ protected boolean dependOnProviderRegistration() {
                                     } else if (configSupplier != null) {
                                         // Try and find a constructor that takes an input stream and use it - or use default constructor and enable deferred initialisation.
                                         try {
-                                            Constructor<? extends Provider> constructor = doPrivileged((PrivilegedExceptionAction<Constructor<? extends Provider>>) () ->  providerClazz.getConstructor(InputStream.class));
-                                            loadedProviders.add(constructor.newInstance(new Object[] { configSupplier.get() }));
-                                        } catch (NoSuchMethodException ignored) {
-                                            Provider provider = providerClazz.newInstance();
-                                            loadedProviders.add(provider);
-                                            deferred.add(provider::load);
+                                            Constructor<? extends Provider> constructor = doPrivileged((PrivilegedExceptionAction<Constructor<? extends Provider>>) () -> providerClazz.getConstructor(InputStream.class));
+                                            loadedProviders.add(constructor.newInstance(new Object[]{configSupplier.get()}));
+                                        } catch (NoSuchMethodException constructorDoesNotExist) {
+                                            try {
+                                                Method configureMethod = doPrivileged((PrivilegedExceptionAction<Method>) () -> providerClazz.getMethod("configure", String.class));
```

Comment:
JDK 9 introduced [configure](https://docs.oracle.com/javase/9/docs/api/java/security/Provider.html#configure-java.lang.String-) method to Provider, and the SunPKCS11 constructor where the configuration was passed before was removed. Let me know if usage of this method here could cause backwards incompatibility.